### PR TITLE
Fix quickly drag elements between nested blocks

### PR DESCRIPTION
**Pull request by v1talii-dev**
### Description
If you quickly drag elements between nested blocks, duplicates of the same element appear.

### CODE REVIEW

File: src/gridstack.ts [hunk 1]
```diff
@@ -2530,7 +2526,6 @@ export class GridStack {
     } else if (node._isExternal) {
       // item came from outside (like a toolbar) so nuke any node info
       delete node.el;
-      delete el.gridstackNode;
```
Comment (adumesny) [inline src/gridstack.ts:2533]: doesn't make sense to delete one and not the other... will have to debug the code for that particular case and make sure it doesn't break other cases..

Comment (v1talii-dev) [reply src/gridstack.ts]: I agree. I noticed errors in the console and problems with blocks not being deleted after dragging. Restored this function.

File: src/gridstack.ts [hunk 2]
```diff
@@ -2111,18 +2111,14 @@ export class GridStack {
...
-        }
 
         // calculate the grid size based on element outer size
         helper = helper || el;
         let w = node.w || Math.round(helper.offsetWidth / cellWidth) || 1;
         let h = node.h || Math.round(helper.offsetHeight / cellHeight) || 1;
 
         // if the item came from another grid, make a copy and save the original info in case we go back there
-        if (node.grid && node.grid !== this) {
+        if (node.grid !== this) {
```
Comment (adumesny) [inline src/gridstack.ts:2121]: still doesn't feel right to me without debuging the issue - if node doesn't have grid, we go through that logic which doesn't feel right...

Comment (v1talii-dev) [reply src/gridstack.ts:2121]: I agree with you. Now it seems to me that the problem is not in this place at all, but in calling the mouseEnter event in mouseOver. If you comment on this line as a definition of the source of the problem, then everything works correctly - https://github.com/gridstack/gridstack.js/blob/master/src/dd-droppable.ts#L135

**Comment by v1talii-dev** [comment]
Redid PR. I discovered a bug of redundant event calling. When the "mouseenter" event was called, the "mouseleave" event was called inside that event. That is, they cleared the previous location of the block and showed a new one. The "mouseleave" event contains the logic for inserting a location into the parent (when we drag the block up into the parent). So, we got a situation where we dragged a field to a child, and the “mouseenter” event was triggered, in turn, “mouseleave” was called, and then “mouseenter” was called. Removed the redundant "mouseenter" step specifically for the program call (the behavior when moving the mouse was preserved).

File: src/dd-droppable.ts [hunk 3]
```diff
@@ -120,7 +120,7 @@ export class DDDroppable extends DDBaseImplement implements HTMLElementExtendOpt
     }
     this.triggerEvent('dropout', ev);
 
-    if (DDManager.dropElement === this) {
+    if (!external && DDManager.dropElement === this) {
```
Comment (adumesny) [inline src/dd-droppable.ts:123]: DDManager.dropElement === this can be removed since on line 113

File: src/dd-droppable.ts [hunk 4]
```diff
@@ -120,7 +120,7 @@ export class DDDroppable extends DDBaseImplement implements HTMLElementExtendOpt
...
       delete DDManager.dropElement;
```
Comment (adumesny) [inline src/dd-droppable.ts:124]: this should still happen. move to line 114.
rename the var fromParent and instead the next comment should be:
// if we're still over a parent droppable, send it an enter as we don't get one from leaving nested children if it's not already calling us
if (!fromParent) {
  let parentDrop: DDDroppable;
  let parent: DDElementHost = this.el.parentElement;
  ....

**Pull request merged**